drivers/video/video_common.c

```diff
@@ -130,19 +130,39 @@ void video_closest_frmival_stepwise(const struct video_frmival_stepwise *stepwis
 	uint64_t step = stepwise->step.numerator;
 	uint64_t goal = desired->numerator;
 
-	/* Set a common denominator to all values */
-	min *= stepwise->max.denominator * stepwise->step.denominator * desired->denominator;
-	max *= stepwise->min.denominator * stepwise->step.denominator * desired->denominator;
-	step *= stepwise->min.denominator * stepwise->max.denominator * desired->denominator;
-	goal *= stepwise->min.denominator * stepwise->max.denominator * stepwise->step.denominator;
+	/* Compute a common denominator product */
+	uint64_t den_product = (uint64_t)stepwise->min.denominator *
+			       (uint64_t)stepwise->step.denominator *
+			       (uint64_t)desired->denominator;
 
-	/* Saturate the desired value to the min/max supported */
+	min *= (uint64_t)stepwise->max.denominator * (uint64_t)stepwise->step.denominator *
+	       (uint64_t)desired->denominator;
+
+	max *= den_product;
+
+	step *= (uint64_t)stepwise->min.denominator * (uint64_t)stepwise->max.denominator *
+		(uint64_t)desired->denominator;
+
+	goal *= (uint64_t)stepwise->min.denominator * (uint64_t)stepwise->max.denominator *
+		(uint64_t)stepwise->step.denominator;
+
+	/* Saturate goal to the supported range */
 	goal = CLAMP(goal, min, max);
 
-	/* Compute a numerator and denominator */
-	match->numerator = min + DIV_ROUND_CLOSEST(goal - min, step) * step;
-	match->denominator = stepwise->min.denominator * stepwise->max.denominator *
-			     stepwise->step.denominator * desired->denominator;
+	/* Prevent division by zero */
+	if (step == 0U) {
+		match->numerator = goal;
+		match->denominator = 1;
+		return;
+	}
```

Comment:
When the step is zero, then this function should not be called as it works only on "stepwise" data with a non-zero step...

An error handling strategy seems more suited IMHO. For instance: `match->numerator = 0;`. Or `return -EINVAL` (would need to change the function type).